src/mixtures/mixturemodel.jl

```diff
@@ -244,11 +244,11 @@ function cov(d::MultivariateMixture)
         pi = p[i]
         if pi > 0.0
             c = component(d, i)
-            # todo: use more in-place operations
-            md = mean(c) - m
-            axpy!(pi, md*md', V)
+            md .= mean(c) .- m
+            BLAS.syr!('U', pi, md, V)
```

Comment:
Should the `pi` be cast to the appropriate type here? I think `BLAS` expects floating-point numbers.

Reply:
In principle, I'd like to avoid BLAS calls even though `md` and `V` are (currently) `Array`s. But it seems `mul!(::Symmetric, ::Vector, ::Adjoint, ...)` is not supported and according to a benchmark on my machine repeated generic `mul!` calls can be slower than `syr!` and a single `copytri!`... Everything apart from `pi` is hardcoded to `Float64` currently though (something to be fixed as well), so
```suggestion
            BLAS.syr!('U', Float64(pi), md, V)
```
should make it safe.

Reply:
Yes, I agree that it is a bit unfortunate that there isn't a higher level API for either of these. This should be fine for now with the change above.

Reply:
Is there a test to cover this line? The fact that the tests passed without this change seems to suggest that this isn't covered.

Reply:
I'll check. I assume that - as in many tests - only distributions with Float64 parameters are tested. Here, eg, tests with a multivariate mixture with mixture probabilities of type Float64 would have passed even without the change.

Reply:
That was indeed the case, it was only tested with `pi::Float64`. I added a test with `Float32` in https://github.com/JuliaStats/Distributions.jl/pull/1967/commits/a065a65a676c58afa1c734b710c49f30a2606ca3 that would error without the `Float64(pi)` change.